demo/src/relying_party_frontend/src/lib/components/BuildConsentMessage.svelte

```diff
@@ -0,0 +1,59 @@
+<script lang="ts">
+	import type { IcpWallet } from '@dfinity/oisy-wallet-signer/icp-wallet';
+	import { isNullish, nonNullish } from '@dfinity/utils';
+	import { fade } from 'svelte/transition';
+	import Button from '$core/components/Button.svelte';
+	import Value from '$core/components/Value.svelte';
+	import { accountsStore } from '$lib/stores/accounts.store';
+	import { authStore } from '$core/stores/auth.store';
+	import type { Icrc1TransferRequest } from '@dfinity/ledger-icp';
+	import { E8S_PER_ICP } from '$core/constants/app.constants';
+
+	type Props = {
+		wallet: IcpWallet | undefined;
+	};
+
+	let { wallet }: Props = $props();
+
+	const INVALID_LEDGER_CANISTER_ID = import.meta.env.VITE_SATELLITE_ID;
+	console.log(INVALID_LEDGER_CANISTER_ID);
+
+	const onclick = async () => {
+		// TODO: handle errors
+		if (isNullish($authStore.identity)) {
+			return;
+		}
+
+		if (isNullish($accountsStore)) {
+			return;
+		}
+
+		const account = $accountsStore?.[0];
+
+		if (isNullish(account)) {
+			return;
+		}
+
+		const request: Icrc1TransferRequest = {
+			to: {
+				owner: $authStore.identity.getPrincipal(),
+				subaccount: []
+			},
+			amount: 1n * (E8S_PER_ICP / 2n)
+		};
+
+		await wallet?.icrc1Transfer({
```

Comment:
should there be an `if` condition beforehand to check if `wallet` is non-nullish? I see that you put it in the html part, but not in this function

Reply:
If the demo were a real app, yes, I would add an if statement and display an error. However, since it is primarily intended for E2E testing, ignoring the fact that the wallet would not be initialized here is acceptable. If the wallet fails to perform any action, the related test would fail to validate the process. Therefore, I think it's fine, but I don't have a strong opinion either way.

Reply:
Ah thanks for the explanation!